### PR TITLE
Add relay network test harness

### PR DESCRIPTION
**Pull request by kriskowal**

### CODE REVIEW

File: node/test/lib/relay_network.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        self.serviceChannels.push(channels);
+        self.serviceChannelsByName[serviceName] = channels;
+    });
+
+    // Create a relay topology for egress nodes.
+    self.topology = {};
+    self.serviceChannels.forEach(function (channels, index) {
+        var serviceName = self.serviceNames[index];
+        var relayHostPorts = [];
+        for (
```
Comment (Raynos) [inline node/test/lib/relay_network.js]: nitpick: for loops like this are weird; please one line or shorter var names.

File: node/test/relay.js [hunk 2]
```diff
@@ -180,3 +181,31 @@ function echo(req, res, arg2, arg3) {
     res.headers.as = 'raw';
     res.sendOk(arg2, arg3);
 }
+
+RelayNetwork.test('relay through a network', {
+    serviceNames: ['alice', 'bob'],
+    numInstancesPerService: 1,
+    numConnections: 1,
```
Comment (Raynos) [inline node/test/relay.js]: Consider using `kValue: 1` instead because that's basically what it is.

File: node/test/lib/relay_network.js [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    });
+
+    // Create and connect service channels
+    self.subChannels = [];
+    self.subChannelsByName = {};
+    self.serviceChannels.forEach(function (channels, serviceIndex) {
+        var serviceName = self.serviceNames[serviceIndex];
+        var subChannels = channels.map(function (channel, channelIndex) {
+            var subChannel = channel.makeSubChannel({
+                serviceName: serviceName,
```
Comment (Raynos) [inline node/test/lib/relay_network.js]: add requestDefaults for `hasNoParent` and `cn` header.

Comment (kriskowal) [inline node/test/lib/relay_network.js]: :heavy_check_mark:

**Comment by kriskowal** [comment]
Feedback fed back. @Raynos @jcorbin

**Comment by Raynos** [comment]
@kriskowal its hard to review when you rebase (lol)

**Comment by Raynos** [comment]
lgtm.

**Pull request merged**